src/stages/models/helper.ts

```diff
@@ -61,10 +63,26 @@ function getCurrentPercentage(stageSummary: StageSummary): number {
   return Math.floor((completed / total) * PERCENTAGE_MULTIPLIER);
 }
 
+/**
+ * This method is used to get the initial xstate snapshot for a stage.
+ * @param stage The stage object containing the initial state flag.
+ * @returns The initial xstate snapshot for the stage, Created as default or Wait if flag is true.
+ */
+function getInitialXstate(stage: StageCreateWithTasksModel): Snapshot<unknown> {
+  const createStageActor = createActor(stageStateMachine).start();
+
+  if (stage.isWaiting ?? false) {
```

Comment:
```suggestion
  if (stage.isWaiting === true) {
```
Nullish coalescing operator (??) for me should be used in assignments, not boolean logic.